### ISSUE TITLE
gunicorn fails to launch

### ISSUE THREAD

**Issue by chivalry** [miguelgrinberg/microblog#153]
I'm working through chapter 17, attempting to launch gunicorn, but it's failing, and I don't understand why. `app/__init__.py` is throwing an error when trying to import from the flask module, but when I do the exact same thing in the python shell, I get no error.

I'm running this using the Vagrant instructions you gave on a macOS host machine.

Here's the input and output I'm getting:

```
(venv) vagrant@ubuntu-xenial:~/microblog$ python
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from flask import Flask, request, current_app
>>> exit()
(venv) vagrant@ubuntu-xenial:~/microblog$ gunicorn -b localhost:8000 -w 4 microblog:app
[2019-03-08 20:23:44 +0000] [11574] [INFO] Starting gunicorn 19.4.5
[2019-03-08 20:23:44 +0000] [11574] [INFO] Listening at: http://127.0.0.1:8000 (11574)
[2019-03-08 20:23:44 +0000] [11574] [INFO] Using worker: sync
[2019-03-08 20:23:44 +0000] [11578] [INFO] Booting worker with pid: 11578
[2019-03-08 20:23:44 +0000] [11578] [ERROR] Exception in worker process:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 515, in spawn_worker
    worker.init_process()
  File "/usr/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 122, in init_process
    self.load_wsgi()
  File "/usr/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 130, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/lib/python2.7/dist-packages/gunicorn/util.py", line 366, in import_app
    __import__(module)
  File "/home/vagrant/microblog/microblog.py", line 1, in <module>
    from app import create_app, db, cli
  File "/home/vagrant/microblog/app/__init__.py", line 4, in <module>
    from flask import Flask, request, current_app
ImportError: No module named flask
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/gunicorn/arbiter.py", line 515, in spawn_worker
    worker.init_process()
  File "/usr/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 122, in init_process
    self.load_wsgi()
  File "/usr/lib/python2.7/dist-packages/gunicorn/workers/base.py", line 130, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/lib/python2.7/dist-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/lib/python2.7/dist-packages/gunicorn/util.py", line 366, in import_app
    __import__(module)
  File "/home/vagrant/microblog/microblog.py", line 1, in <module>
    from app import create_app, db, cli
  File "/home/vagrant/microblog/app/__init__.py", line 4, in <module>
    from flask import Flask, request, current_app
ImportError: No module named flask
[2019-03-08 20:23:44 +0000] [11578] [INFO] Worker exiting (pid: 11578)
[2019-03-08 20:23:44 +0000] [11574] [INFO] Shutting down: Master
[2019-03-08 20:23:44 +0000] [11574] [INFO] Reason: Worker failed to boot.
```

How can the python shell be able to import the flask module while the microblog app can't?

**Comment by chivalry** [comment]
I think I figured it out.

The book doesn't mention (or I can't find it) how to install gunicorn. I used `sudo apt-get` and when that ran, I guess it wasn't running within the virtual environment. I uninstalled that version, made sure I was in the virtual environment, and then ran `pip install gunicorn`. I still had a problem with the path (it was looking for gunicorn at `/usr/bin/gunicorn`), but that seems to have been fixed by deactivating and reactivating the virtual environment.
